### PR TITLE
Change feedback page and form, refactor validation

### PR DESCRIPTION
**Pull request by AndrewHelix**
1. Refactor validation. Now I can add new rules for validation and not use previous ugly code 
Like it was
`let selectedInput: ValidateElementInfoType;`
    `if (elementName === 'name') {`
     ` selectedInput = name;`
`    }`
 `   if (elementName === 'email') {`
 `     selectedInput = email;`
  `  }`
   ` selectedInput!.validateInput(elementName, elementValue);`

For every input I need to add new check. Now I don't need to do it. Only add validation rules for new input.

I hope your "custom validation" mean smth like I done:)

2. Change feedback UI. Remove bad rating input and add stars (saw at Auryte and it more beautiful than was)
3. Added previous feedbacks in feedback page
4. Added a bit changes in feedback form UI 
Screenshot 2022-09-29 173417

### CODE REVIEW

File: src/components/feedback/Feedback.tsx [hunk 1]
```diff
@@ -73,94 +70,105 @@ export const Feedback: FC = () => {
...
+            className={styles['feedback-input']}
+            onChange={validatedInputHandler}
+            placeholder="required field"
+            required
+          />
+          {!isValid.name && state.name.length ? (
+            <span className={styles['validation-error']}>
+              The name must include letters and numbers and be longer than 4 characters
+            </span>
+          ) : null}
```
Comment (JuliaShavyrova) [inline src/components/feedback/Feedback.tsx]: no need to use ternary operator here, just use `&&`. also no need to write `.length`, `!!state.name` is enough

File: src/components/feedback/ViewFeedbacks.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <section className={styles['view-feedback']}>
+      <h2 className={styles['feedbacks-section-title']}>Feedbacks</h2>
+      {feedbacks.map(feedback => (
+        <article className={styles['feedbacks-article']} key={feedback.id}>
+          <h3 className={styles['feedback-article-name']}>{feedback.name}</h3>
+          <div className={styles['feedback-stars']}>
+            {[1, 2, 3, 4, 5].map(value => (
+              <React.Fragment key={value}>
+                {value <= feedback.rating ? <IoMdStar /> : <IoMdStarOutline />}
```
Comment (JuliaShavyrova) [inline src/components/feedback/ViewFeedbacks.tsx]: you can reuse `RatingStar` component, or create separate component for rating and make it suitable both for edit and view modes

File: src/hooks/useValidate.ts [hunk 3]
```diff
@@ -1,22 +1,44 @@
...
+  email: (value: string) => REG_EXP_FOR_EMAIL.test(value),
+  phone: (value: string) => REG_EXP_FOR_PHONE.test(value)
 };
 
-export default function useValidate(): ValidateElementInfoType {
-  const [isValid, setIsValid] = useState<boolean>(false);
+export default function useValidate(settings: {
+  [index: string]: string;
+}): ValidateElementInfoType {
+  const [isValid, setIsValid] = useState<{ [index: string]: boolean }>({
```
Comment (JuliaShavyrova) [inline src/hooks/useValidate.ts]: `{ [index: string]: boolean }` can be defined as interface

**Comment by JuliaShavyrova** [comment]
also please pay attention on design: submit button doesn't look disabled for empty form, and button text is almost not visible

**Comment by AndrewHelix** [comment]
Invalid and empty form
Screenshot 2022-09-30 145520

Valid form
Screenshot 2022-09-30 145549

File: src/components/feedback/Feedback.tsx [hunk 4]
```diff
@@ -1,188 +1,28 @@
...
+import { ViewFeedbacks } from './ViewFeedbacks';
+import { FeedbackFrom } from './FeedbackForm';
 import { FeedbackState } from 'types/feedbackType';
-import styles from './Feedback.css';
-import useValidate from 'hooks/useValidate';
 import { sendFeedback } from 'services/localStorage';
-import classNames from 'classnames';
-import { RatingStar } from './RatingStar';
-import { ViewFeedbacks } from './ViewFeedbacks';
+import { getFeedbacks } from 'services/localStorage';
```
Comment (JuliaShavyrova) [inline src/components/feedback/Feedback.tsx]: import { sendFeedback } from 'services/localStorage';
import { getFeedbacks } from 'services/localStorage';

please combine import

File: src/components/feedback/RatingStar.tsx [hunk 5]
```diff
@@ -5,10 +5,18 @@ import styles from './Feedback.css';
...
+  size?: number;
+  onChange?: (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
 };
 
-export const RatingStar: FC<RatingStarProps> = ({ onChange, ratingValue, currentRating }) => {
+export const RatingStar: FC<RatingStarProps> = ({
+  onChange,
+  ratingValue,
+  currentRating,
+  cssClassName,
```
Comment (JuliaShavyrova) [inline src/components/feedback/RatingStar.tsx]: please rename `cssClassName` to `className`

File: src/components/feedback/RatingStar.tsx [hunk 6]
```diff
@@ -19,8 +27,8 @@ export const RatingStar: FC<RatingStarProps> = ({ onChange, ratingValue, current
         onChange={onChange}
         className={styles['rating-input']}
       />
-      <label htmlFor={`rating${ratingValue}`} className={styles['rating-label']}>
-        {currentRating >= ratingValue ? <IoMdStar size={30} /> : <IoMdStarOutline size={30} />}
+      <label htmlFor={`rating${ratingValue}`} className={styles[cssClassName!]}>
```
Comment (JuliaShavyrova) [inline src/components/feedback/RatingStar.tsx]: no need to create enum for star className and incapsulate custom style inside this component, you can just use className={className} and define additional styles in parent component

File: src/components/feedback/RatingStar.tsx [hunk 7]
```diff
@@ -14,9 +14,10 @@ export const RatingStar: FC<RatingStarProps> = ({
   onChange,
   ratingValue,
   currentRating,
-  cssClassName,
+  className,
   size
 }) => {
+  console.log(className);
```
Comment (JuliaShavyrova) [inline src/components/feedback/RatingStar.tsx]: please remove

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**